src/System.Private.CoreLib/src/System/Threading/Interlocked.cs

```diff
@@ -59,17 +60,17 @@ public static long Decrement(ref long location)
         public static extern double Exchange(ref double location1, double value);
 
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        public static extern object Exchange(ref object location1, object value);
+        public static extern object? Exchange(ref object? location1, object? value);
```

Comment:
Why is location1 nullable? I see that the native implementation throws if location1 == `NULL`

https://github.com/dotnet/coreclr/blob/4932d0f6fa87b9a6e3293aacd35cecedde44d87a/src/vm/comutilnative.cpp#L1863

Reply:
It's not the ref that's nullable, it's the thing the ref references that's nullable.

Reply:
Ah right...